components/Input/Input.jsx

```diff
@@ -89,11 +89,26 @@ class Input extends React.Component {
   constructor(props) {
     super(props);
 
-    const { type, id } = props;
+    const { type, id, value } = props;
 
-    this.state = { type };
+    this.state = {
+      type,
+      currentValue: value,
+    };
 
     this._id = id || ID_GENERATOR.next().value;
+
+    this.onChangeInput = this.onChangeInput.bind(this);
```

Comment:
Why not use [public class fields](https://github.com/tc39/proposal-class-fields) instead of binding functions?

This project adopts this pattern to make the code clean and there is no reference of binding events/methods/functions in any source code.

Reply:
@ggdaltoso fixed in last commit.